backend/src/test/kotlin/com/infowings/catalog/auth/user/UserInitTest.kt

```diff
@@ -0,0 +1,101 @@
+package com.infowings.catalog.auth.user
+
+import com.infowings.catalog.AbstractDatabaseTest
+import com.infowings.catalog.storage.transaction
+import io.kotlintest.inspectors.forExactly
+import io.kotlintest.shouldBe
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertAll
+import org.springframework.beans.factory.annotation.Autowired
+
+@Suppress("UnsafeCallOnNullableType")
+class UserInitTest : AbstractDatabaseTest() {
+
+    @Autowired
+    lateinit var userDao: UserDao
+
+    @Test
+    fun `Database initializer should not change the database if properties do not have new user`() {
+        val userInitializer = UserDatabaseInitializer(orientDatabase, userDao)
+        val userVertexes = transaction(orientDatabase) {
+            userDao.getAllUserVertices().toList()
+        }
+        val firstUserVertex = userVertexes.first()
+
+        val userProperties = UserProperties().apply {
+            user.add(createUserConfig(firstUserVertex.username, firstUserVertex.password, firstUserVertex.role))
+        }
+
+        userInitializer.initUsers(userProperties)
+
+        val firstUserVertexAfterInit = transaction(orientDatabase) {
+            userDao.findByUsername(firstUserVertex.username)!!
+        }
+
+        assertAll(
+            "All fields before and after initialization should be equal",
+            { firstUserVertex.username shouldBe firstUserVertexAfterInit.username },
+            { firstUserVertex.password shouldBe firstUserVertexAfterInit.password },
+            { firstUserVertex.role     shouldBe firstUserVertexAfterInit.role }
+        )
```

Comment:
It makes sense to check that vertex id is the same. Because if new vertex is created with the same user info it will not be caught by current checks howover it will affect history collection in destructive way.